lib/rubocop/cop/rspec/leading_subject.rb

```diff
@@ -54,13 +54,17 @@ def check_previous_nodes(node)
         private
 
         def offending_node(node)
-          node.parent.each_child_node.find do |sibling|
+          parent(node).each_child_node.find do |sibling|
             break if sibling.equal?(node)
 
             yield sibling if offending?(sibling)
           end
         end
 
+        def parent(node)
+          node.each_ancestor(:block).first.body
```

Comment:
@pirj Could you help me understand _how_ this fixed the problem? (Extra points for adding the explanation to the commit message ❤️)

Reply:
Sure. Described what was wrong here: https://github.com/rubocop-hq/rubocop-rspec/pull/1011/commits/87562acddffee61d98d90858e24b724c47e47b9f

By picking the nearest outer block body, we make sure we only pick real siblings inside that block.